### PR TITLE
List-n to display pinned and unpinned notes separately, catch null point exception for find-n

### LINKED ISSUE #48: Logging
Logging can be useful for troubleshooting problems. A good logging system records some system information regularly. When an unanticipated failure occurs, their associated log files may provide indications of what went wrong and action can then be taken to prevent it from happening again.

### PR DESCRIPTION
**Pull request by R-Ramana**
Closes #48 #62 

Add the following functionality
- List-n can now display notes that are tagged as pinned. Can sort (un)/pinned notes as well.

Fixed a bug which caused the program to terminate
- Catch null point exception for find-n

Included the following additional code that is nice to have
- Added a setUpLogger() method for whoever wants to do logging.
- Added logging to find-n

### CODE REVIEW

File: src/main/java/seedu/duke/command/ListNoteCommand.java [hunk 1]
```diff
@@ -89,24 +90,49 @@ public ListNoteCommand(Boolean isAscendingOrder, ArrayList<String> tags) {
...
+                unpinnedNotes.add(notebook.getNotes().get(i));
+            }
+        }
 
         ArrayList<Note> sortedNotes = (ArrayList<Note>) notebook.getNotes().stream()
                 .filter(Objects::nonNull)
                 .sorted(Comparator.comparing(a -> a.getTitle().toLowerCase()))
                 .collect(Collectors.toList());
 
         if (tags == null) {
```
Comment (Nazryl) [inline src/main/java/seedu/duke/command/ListNoteCommand.java:113]: Maybe can give a short comment on what's happening here.

File: src/main/java/seedu/duke/util/Parser.java [hunk 2]
```diff
@@ -485,7 +492,17 @@ private Command prepareDeleteNote(String userMessage) throws SystemException {
      * @throws SystemException for missing keyword.
      */
     private Command prepareFind(String userMessage) throws SystemException {
```
Comment (Nazryl) [inline src/main/java/seedu/duke/util/Parser.java:494]: @param?

File: src/main/java/seedu/duke/command/ListNoteCommand.java [hunk 3]
```diff
@@ -180,37 +199,40 @@ public String execute() {
...
-                        .append(SUFFIX_INDEX)
-                        .append(sortedNotes.get(i).getTitle())
-                        .append(" ")
-                        .append(sortedNotes.get(i).getTagsName())
-                        .append(InterfaceManager.LS);
-            }
-            noteString = noteStrBuilder.toString();
+    /**
+     * Method compiles the ArrayList items and appends the items to a String.
+     *
```
Comment (Nazryl) [inline src/main/java/seedu/duke/command/ListNoteCommand.java:222]: @param here too

File: src/main/java/seedu/duke/command/ListNoteCommand.java [hunk 4]
```diff
@@ -180,37 +199,40 @@ public String execute() {
...
-                j++;
-            }
-            noteString = noteStrBuilder.toString();
+            Collections.reverse(sortedNotes);
+            noteStrBuilder = getNoteString(sortedNotes);
+            //noteString = noteStrBuilder.toString();
 
         } else if (isAscendingOrder) {
+            noteStrBuilder = getNoteString(sortedNotes);
+            //noteString = noteStrBuilder.toString();
```
Comment (Nazryl) [inline src/main/java/seedu/duke/command/ListNoteCommand.java:215]: Forget to remove? Or just ignore for next PR use.

**Review by Nazryl — APPROVED** [review]
Good job very clean.

File: src/main/java/seedu/duke/util/Parser.java (hunk 2 above)
Comment (R-Ramana) [reply src/main/java/seedu/duke/util/Parser.java:494]: Already in the code

**Pull request merged**